firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -142,17 +142,17 @@ public Thread newThread(Runnable r) {
    * FirebaseOptions#getApplicationId()} , and ({@link FirebaseOptions#getProjectId()} are not null
    * or empty.
    */
-  private void preConditionChecks() {
-    Preconditions.checkNotEmpty(getApplicationId());
-    Preconditions.checkNotEmpty(getProjectIdentifier());
-    Preconditions.checkNotEmpty(getApiKey());
```

Comment:
Did you remove these methods?

getApplicationId, getProjectIdentifier, getApiKey

Also, for getProjectIdentifier we wanted to either use project id or sender id.

Reply:
- No, I didn't remove those methods. I just made preConditionChecks to be static and move it to constructor from getId() and getToken().

- getProjectIdentifier was changed to take only projectId on last PR as requested by Rayo.

Reply:
nit: If those methods arent used remove them.

Reply:
Thanks, those methods are used somewhere else. So I keep them.